### DISCUSSION TITLE
Deliver a Symfony local app

### DISCUSSION THREAD

**Discussion by PierreLouisPAUGAM** [symfony/symfony#43159] [Q&A]
Hello,

I'm a student dev in a company and I will soon have to develop an Symfony app which will have to be able to run locally (without any internet connexion).

During the development I will be asked to regularly deliver versions of the app to future users, these users won't necessarily have any dev background so the delivery will have to be : 
    - Copying / Pasting a folder from an USB stick
    - Double clicking a single file inside that folder
    - > Boom ! The application opens in a web browser.

My question is : what would be the cleanest way to do so ? I can't previoulsy install a WAMP on the user's computer. I can't put the folder in a specific place on their computer neither. I need the app to work "autonomously", like a portable app.

Thanks in advance :)

**Comment by stof** [comment]
Well, a Symfony app is useless without a PHP runtime and a webserver. That's what you need to make "autonomous" first.

**Reply by PierreLouisPAUGAM** [reply]
Thanks for your answer, I'm quite new to Symfony so I'm still reading through it's documentation. I just found this : https://symfony.com/doc/current/setup/symfony_server.html

Would that not be an "autonomous" PHP runtime and webserver bundled directly in Symfony ? If so it may be possible to include a single .bat file with the app delivery which would start the server and run the app.

**Reply by stof** [reply]
The `symfony` CLI contains a webserver. But it still relies on having a PHP runtime installed.

**Reply by PierreLouisPAUGAM** [reply]
Thanks.

So a solution would be to put the project into a portable WAMP and to find a way to start its services and launch the app through a single .bat file. I'm going to try and do this :)
